Let logging to file respect log level

hubertp opened the pull request:
### Pull Request Description

This change fixes a regression introduced in #7918, which prevented the execution from setting the right log level either via env var or parameter.

Now passing either of the options returns logs of the expected level in the log file:
- `ENSO_LOG_TO_FILE_LOG_LEVEL = trace`
- ... `-vv` ...

Fixes #8274 

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

hubertp committed:
Let logging to file respect log level

This change fixes a regression introduced in #7918, which prevented the
execution from setting the right log level either via env var or
parameter.

Now passing either of the options returns logs of the expected level
in the log file:
- `ENSO_LOG_TO_FILE_LOG_LEVEL = trace`
- ... `-vv` ...

Akirathan commented on engine/language-server/src/main/resources/application.conf:
```diff
@@ -37,9 +37,9 @@ logging-service {
-    #org.enso.languageserver.protocol.json.JsonConnectionController = debug
-    #org.enso.jsonrpc.JsonRpcServer = debug
-    #org.enso.languageserver.runtime.RuntimeConnector = debug
+    org.enso.languageserver.protocol.json.JsonConnectionController = debug
```
Nice to see that this is uncommented by default.

hubertp replied:
Yes, I got fed up with it. In `trace` you can't see anything because of it.

Akirathan reviewed (changes requested):
When I use
```diff
diff --git a/engine/language-server/src/main/resources/application.conf b/engine/language-server/src/main/resources/application.conf
index c2d9e1ded..eb6ac846c 100644
--- a/engine/language-server/src/main/resources/application.conf
+++ b/engine/language-server/src/main/resources/application.conf
@@ -56,7 +56,7 @@ logging-service {
   default-appender = socket
   default-appender = ${?ENSO_APPENDER_DEFAULT}
   log-to-file {
-    enable = false
+    enable = true
     enable = ${?ENSO_LOG_TO_FILE}
     log-level = debug
     log-level = ${?ENSO_LOG_TO_FILE_LOG_LEVEL}
```
I still have 5 GBs of output in my stdout, even though the default log level to console should be INFO. What is the purpose of `log-to-file.enable` property then? 

Note that I like to set the `log-to-file.log-level = trace` in `application.conf` rather than via env variable. That is why I also tried to set `log-to-file.enable = true`.

hubertp committed:
Ensure context is reset with Socketappender

It makes little sense in the current environment to log to file while
also logging to socket, because the latter most of the time is doing
that anyway.
This change prevents from flodding console with logs.

It would make sense to support such feature if language server was
started independently (without `project-manager`).

Left out the section in language server's `application.conf` in case
someone wants to use an appender different from socket one.

hubertp committed:
more info

hubertp commented:
> When I use
> ... 
> I still have 5 GBs of output in my stdout, even though the default log level to console should be INFO. What is the purpose of ?`log-to-file.enable` property then?

It's a problem unrelated to the fact what this PR is fixing but I've been thinking how to best tackle this anyway.

As noted internally setting log-to-file for language server that is being started by project-manager makes little sense when using socket appender. It probably would make sense if language server was started separately. For now, just hardcoded the fact that language server should always reset the `LoggerContext` thus ensuring console is not flooded by logs. I would only spend more time on this if there is a real use-case to support it.
